frontend/syntax.asdl

```diff
@@ -298,12 +298,12 @@ module syntax
   | ExpandedAlias(command child, List[Redir] redirects, List[EnvPair] more_env)
   | Sentence(command child, Token terminator)
     # Note: Only represents "bare assignment"
-  | ShAssignment(List[AssignPair] pairs, List[Redir] redirects)
+  | ShAssignment(Token var, List[AssignPair] pairs, List[Redir] redirects)
   | ControlFlow(Token keyword, word? arg_word)
     # Note: There are spids for every pipeline operator, parallel to
     # stderr_indices
-  | Pipeline(List[command] children, bool negated, List[int] stderr_indices)
-  | AndOr(List[id] ops, List[command] children)
+  | Pipeline(Token pipline_op, List[command] children, bool negated, List[int] stderr_indices)
+  | AndOr(Token op, List[id] ops, List[Token] op_toks, List[command] children)
```

Comment:
I think we can remove `List[id]` and just have `List[Token] ops`

A token contains an Id